www/scripts/codecheckerviewer/hooks.js

```diff
@@ -0,0 +1,32 @@
+// -------------------------------------------------------------------------
+//                     The CodeChecker Infrastructure
+//   This file is distributed under the University of Illinois Open Source
+//   License. See LICENSE.TXT for details.
+// -------------------------------------------------------------------------
+define([],
+function () {
+  // This module exports a dummy interface showing what hooks are wired in
+  // to the usage of CodeChecker. This interface can be used to fine-tune a
+  // local CodeChecker install to execute arbitrary extra tasks when a
+  // certain action is triggered.
+
+  return {
+    runsListed : function (numRuns) {
+    },
+
+    runObserved : function (run) {
+    },
+
+    bugFilteringChanged : function (numFilters) {
+    },
+
+    bugFilteringChangedDiff : function (numFilters) {
+    },
+
+    bugReportOpened : function (report) {
+    },
+
+    bugPathEventClicked : function (index, bugPathEvent) {
+    }
+  };
+});
```

Comment:
This should be implemented by `dojo/topic` which provides this message sending functionality. If you browse the source code, you can find `codechecker/...` topics already registered. And personally I think that this interface shouldn't be added to the code base without implementation just for future use.